### DISCUSSION TITLE
FileZilla uploading multiple big files parallel

### DISCUSSION THREAD

**Discussion by SebastianKuesters** [QuorumDMS/ftp-srv#176] [Q&A]
Hi,

I have found an issue when I am try to modify the default sample script:

What I have done?
I have created an nodejs server using the following server.js code, which I have take from the samples:

```
const bunyan = require('bunyan');
const FtpSrv = require('./src');
const FileSystem = require('./src/fs');

const server = new FtpSrv({
  url: 'ftp://0.0.0.0:21',
  log: bunyan.createLogger({
    name: 'debug',
    level: 'trace',
    streams: [
      {
        level: 'debug',
        stream: process.stdout
      }
    ]
  }),
  anonymous: true,
  greeting: ['Hello Jong', 'Wie gehts?']
});

server.on('login', (data, resolve, reject) => {
  resolve({ fs: new FileSystem(data.connection, undefined) });
});
server.listen();

```

What is the problem?
When I connect 8 clients at once to my FTP server which runs in an Ubuntu server on Azure and every client starts to upload a file bigger than 300MB then for most uploads will stuck at 100% in FileZilla and run into a timeout:

```
22:30:13	Befehl:	STOR Docker Desktop Installer - Kopie.exe
22:30:13	Antwort:	150 File status okay; about to open data connection
22:30:13	Fehler:	Zeitüberschreitung der Verbindung nach 20 Sekunden Inaktivität // Timeout after 20 seconds 
22:30:13	Fehler:	Dateiübertragung fehlgeschlagen nach Übertragung von 876.801.240 Bytes in 613 Sekunden
22:30:13	Status:	Verbindung zum Server getrennt // server connection closed
```

I would be very happy if you could help me with this issue!

Best,
Sebastian

**Comment by trs** [comment]
There must be some fundamental reason why large files fail.
I wonder if there is an incorrect timeout or something?
This will take more investigating.
